benchmarks/benchmarks/sparse_linalg_svds.py

```diff
@@ -1,31 +1,62 @@
 import os
+import warnings
 import numpy as np
 from .common import Benchmark, safe_import
 
 with safe_import():
+    from scipy.linalg import svd
     from scipy.sparse.linalg import svds
 
+# ensure that we are benchmarking a consistent outcome;
+# (e.g. if the code wasn't able to find a solution accurately
+# enough the timing of the benchmark would become useless).
+msg = ("the benchmark code did not converge as expected, "
+       "the timing is therefore useless")
+
 
 class BenchSVDS(Benchmark):
-    # Benchmark SVD using the MatrixMarket test matrices recommended by the
+    # Benchmark SVDS using the MatrixMarket test matrices recommended by the
     # author of PROPACK at http://sun.stanford.edu/~rmunk/PROPACK/
+    # 'arpack' convergence uniformly for all singular values, while
+    # 'lobpcg' extreme singular values may converge much faster, so
+    # the assert is only evaluated on the top k/2 singular values.
+    # The assert uses the relative error since for some tested matrices
+    # the maximal singular values are very large due to poor matrix scaling.
+    # The `maxiter` and `tol` paremeters of `svds` are tuned for fair comaprison.
+    # The dense SVD solve is benchmaked as the base. 
     params = [
-        [25],
+        [20],
         ["abb313", "illc1033", "illc1850", "qh1484", "rbs480a", "tols4000",
          "well1033", "well1850", "west0479", "west2021"],
         # TODO: re-include propack
-        ['arpack', 'lobpcg']  # 'propack' failing (Aug. 2023)
+        ['arpack', 'lobpcg', 'svd']  # 'propack' failing (Aug. 2023)
     ]
     param_names = ['k', 'problem', 'solver']
 
-    def setup(self, k, problem, solver):
+    def __init__(self):
         dir_path = os.path.dirname(os.path.realpath(__file__))
         datafile = os.path.join(dir_path, "svds_benchmark_files",
                                 "svds_benchmark_files.npz")
-        matrices = np.load(datafile, allow_pickle=True)
-        self.A = matrices[problem][()]
+        self.matrices = np.load(datafile, allow_pickle=True)  
+
+    def setup(self, k, problem, solver):
+        if solver == 'svd' and problem == "tols4000":
+            # skip: fails breaking the benchmark
+            raise NotImplementedError()
+
+        self.A = self.matrices[problem][()]
+        _, s, _ = svd(self.A.toarray(), full_matrices=False)
+        self.top_singular_values = np.flip(s[:int(k/2)])
+        self.tol = k * np.max(self.A.shape) * np.finfo(float).eps
+        self.rng = np.random.default_rng(0)
 
     def time_svds(self, k, problem, solver):
-        # consider k = int(np.min(self.A.shape) * k)
-        np.random.seed(0)
-        svds(self.A, k=k, solver=solver)
+        if solver == 'svd':
+            _ = svd(self.A.toarray(), full_matrices=False)
```

Comment:
maybe add a comment that svd doesn't have an argument `k`? Or actually, do we really need to add benchmarks for svd (are there other ones already), beyond using it to get a baseline result for the other solvers?

If we do want to add benchmarks for `svd` specifically, we should either split it out into a separate benchmark (as indicated also by the divergence between the code doing the actual benchmarking here between `svd` and not-`svd`), or note at least in the parameter definition that we're only avoiding redundancy because we currently have a single `k` being benchmarked.

Reply:
My goal here is to add `svd` only as a baseline result for the other solvers. I have not checked other `svd` benchmarks.

Reply:
I have added a clarifying comment as requested